resolver.go

```diff
@@ -26,6 +26,7 @@ func NewTypeResolver() *TypeResolver {
 	r.Register(string(Int), int16(0))
 	r.Register(string(Int), int32(0))
 	r.Register(string(Int), int(0))
+	r.Register(string(Long), int(0))
```

Comment:
Should this only be for 64 but systems?

Reply:
I don't think so, let me explain.

In case of encoding, you can encode `int32` and `int64` as `long`, both should be fine. And this is enabled now because `TypeResolver.Name(int)` will return `long` as a possible type for encoding `int`.

Now, in the case of decoding it's only safe to decode `long` into `int64`. This is till the case though, because in the next line we register `int64` as `long`, so `TypeResolver.Type("long")` will still return `int64`, as it was before.

Reply:
Fair point